### ISSUE TITLE
Accessibility: the "ready to try the new Exercism" modal dialog is not accessible

### ISSUE THREAD

**Issue by saxomagic** [exercism/exercism#6340]
After logging in successfully after not having used Exercism for some time, you are presented with a kind of modal dialog that is not accessible to a screen reader, and requires assistance to click "continue" or similar - after that all is fine.

**Comment by SleeplessByte** [comment]
Thanks for reporting. 

@iHiD tell me if you want pointers on this.

**Comment by iHiD** [comment]
@saxomagic Thanks for reporting.
@ErikSchierboom Could you take this one on and chat to @SleeplessByte as part of it pls.

**Comment by ErikSchierboom** [comment]
I'll get in contact

**Comment by SleeplessByte** [comment]
Erik and I spoke. I'll briefly list the points I've given.

## Links

- [WAI ARIA practices: Dialog Modal](https://www.w3.org/TR/wai-aria-practices/#dialog_modal)
- You probably  want [WAI ARIA practices: Alert Dialog](https://www.w3.org/TR/wai-aria-practices/#alertdialog)
- [Example implementation of Alert dialog](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/alertdialog.html)
- [Pragmatic implementation of dialogs](https://www.accessibility-developer-guide.com/examples/widgets/dialog/)

## Important pointers

1. You want to set the `role`
2. You want to add `aria-describedby` and should add `aria-labelledby` to the dialog
3. You MUST trap the focus, and preferably wrap focus around
4. Tab order should yield Cancel/Close before any other action
5. Escape should call Cancel/Close

## Other points

- Inside the modal you want to add `role="document"`, preferably wrapped around everything, but inside `role="xxxdialog"`. This is to allow screenreaders to switch back to reading-mode inside the dialog.
- For modals (like the one we discussed) should not be able to interact with anything behind the modal.
- Don't be tempted to use `<dialog>` cuzzzz support
- Don't be tempted to leave out a heading. If there is no visual reason for a heading, because it's derived from where you clicked before, add an `sr-only` invisible one. I recommend `h1` or `h2` for modals, and `hx` for non-modals. Should make sense in the document outline.
- Move the focus back to the button that opened the dialog, unless there is none.

**Comment by iHiD** [comment]
Let's remove this dialog altogether now (obvs DJ's comments are probably relevant for other modals etc!)
